packages/runtime/src/next/client.ts

```diff
@@ -652,45 +673,36 @@ export class Makeswift {
   async getComponentSnapshot(
     id: string,
     {
-      siteVersion = MakeswiftSiteVersion.Working,
+      siteVersion: siteVersionPromise,
       locale,
-    }: {
-      siteVersion?: MakeswiftSiteVersion
-      locale?: string
-    } = {},
+    }: { siteVersion: MakeswiftSiteVersion | Promise<MakeswiftSiteVersion>; locale?: string},
   ): Promise<MakeswiftComponentSnapshot> {
     const searchParams = new URLSearchParams()
     if (locale) searchParams.set('locale', locale)
 
+    const siteVersion = await siteVersionPromise
     const response = await this.fetch(
-      `v1/element-trees/${id}?${searchParams.toString()}`,
+      `v2/element-trees/${id}?${searchParams.toString()}`,
       siteVersion,
     )
 
     // If the element tree is not found, we generate a document with null data
     if (response.status === 404) {
-      const apiKeyPrefix = z.string().parse(this.apiKey.split('-').at(0))
-
-      return {
-        document: {
-          id,
-          locale: locale ?? null,
-          data: null,
-          // We pass the seed to generate the elementKey to make sure it's unique for each site.
-          seed: apiKeyPrefix,
-        },
-        cacheData: CacheData.empty(),
-      }
+      return this.getSnapshotWithFallbackDocument(id, locale)
     }
 
     const json = await response.json()
 
     if (!response.ok) {
-      console.error('Failed to get page snapshot', json)
-      throw new Error(`Failed to get page snapshot with error: "${response.statusText}"`)
+      console.error('Failed to get component snapshot', json)
+      throw new Error(`Failed to get component snapshot with error: "${response.statusText}"`)
     }
 
-    const document = makeswiftComponentDocumentSchema.parse(json)
+    const elementTreeResponse = getMakeswiftComponentDocumentSchemaV2.parse(json)
+
+    // favor picking 'elementTree' if it is available.
```

Comment:
Super mega nit incoming...

My stance on comments is that if they describe what the code is doing, they're unnecessary. Sometimes they might even become outdated and cause confusion. I think comments should be generally reserved to explain "why". If we're having to explain "what" with comments then it's a sign we can factor the code in a better way.

Reply:
definitely agree, removed this and on other